In [1]:
import pandas as pd

In [2]:
data = pd.read_csv('ingredientes_estandarizados.csv')

In [3]:
df = data.copy()

In [16]:
df

receta  \
0                  Iogurt grec amb fraules   
1                  Iogurt grec amb fraules   
2                  Iogurt grec amb fraules   
3                  Iogurt grec amb fraules   
4                  Iogurt grec amb fraules   
...                                    ...   
1058  Pollastre al forn a les fines herbes   
1059  Pollastre al forn a les fines herbes   
1060  Pollastre al forn a les fines herbes   
1061  Pollastre al forn a les fines herbes   
1062          Broqueta de gambes amb xoriç   

                    ingrediente_original  cantidad unidad ingrediente_estandar  
0                       Iogurt grec 200g     200.0      g          iogurt grec  
1                       Fraules 100-150g     150.0      g              fraules  
2                    Nous i ametlles 30g      30.0      g      nous i ametlles  
3                             Xía 10-20g      20.0      g                  xia  
4                       Canyella al gust       NaN    NaN             canyella  
...                                  ...       ...    ...                  ...  
1058    5g d'orenga fresc (0.5g en pols)       NaN    NaN         orenga fresc  
1059               15g de mostassa Dijon       NaN    NaN       mostassa dijon  
1060  15ml d'oli d'oliva (una cullerada)       NaN    NaN          oli d'oliva  
1061                 Sal i pebre al gust       NaN    NaN          sal i pebre  
1062                          llagostins     250.0      g           llagostins  

[1063 rows x 5 columns]

In [5]:
df.columns

Index(['receta', 'ingrediente_original', 'cantidad', 'unidad',
       'ingrediente_estandar'],
      dtype='object')

In [6]:
df.shape

(1063, 5)

In [7]:
df['ingrediente_estandar'].nunique()

246

In [115]:
df['ingrediente_estandar'].value_counts().head(28)

ingrediente_estandar
sal i pebre             101
oli d'oliva              86
ous                      50
alls                     45
llimones                 26
mantega sense sal        24
ceba blanca              19
formatge parmesa         18
mantega                  16
ceba morada              14
nata liquida             14
julivert                 14
tomaquets cherry         14
alvocat                  13
eritritol                12
tomaquet                 12
pernil salat             12
cibulet                  11
espinacs                 11
ruca                     11
pebre vermell dolc        9
vinagre                   9
pastanaga                 9
mostassa dijon            9
xampinyons                9
ceba                      8
salmo fumat               8
essència de vainilla      8
Name: count, dtype: int64

In [9]:
mask = df[(df['ingrediente_estandar'].str.contains("dents", na=False)) & (df['cantidad'] == 3)]

In [112]:
mask = df[df['ingrediente_estandar'].str.contains("essència", na=False)]

In [113]:
df.loc[mask.index, 'ingrediente_estandar'] = 'essència de vainilla'
#df.loc[mask.index, 'cantidad'] = None
#df.loc[mask.index, 'unidad'] = None

In [122]:
df.loc[[802],'ingrediente_estandar'] = 'formatge'
#df.loc[[645],'cantidad'] = 1
#df.loc[[781],'unidad'] = None

In [114]:
df[df['ingrediente_estandar'].str.contains("essència", na=False)]

receta         ingrediente_original  \
190  Pa de plàtan amb xispes de xocolata  3 ml d'essència de vainilla   
264                              Granola   5ml d'essència de vainilla   
400                           Ceto-creps   5ml d'essència de vainilla   
409                         Ceto-nutella   5ml d'essència de vainilla   
470      Muffins de nabius (sense sucre)   2ml d'essència de vainilla   
489                 Coquitos de xocolata   1ml d'essència de vainilla   
573                   Brioix de xocolata   2ml d'essència de vainilla   
679                          Cetogranola   5ml d'essència de vainilla   

     cantidad unidad  ingrediente_estandar  
190       NaN    NaN  essència de vainilla  
264       NaN    NaN  essència de vainilla  
400       NaN    NaN  essència de vainilla  
409       NaN    NaN  essència de vainilla  
470       NaN    NaN  essència de vainilla  
489       NaN    NaN  essència de vainilla  
573       NaN    NaN  essència de vainilla  
679       NaN    NaN  essència de vainilla

n_fila = [{
    'receta': 'Broqueta de gambes amb xoriç',
    'ingrediente_original': 'llagostins',
    'cantidad': 250,
    'unidad': 'g',
    'ingrediente_estandar': 'llagostins'
}]

df = pd.concat([df, pd.DataFrame(n_fila)], ignore_index=True)

In [148]:
df.sort_values(by=['ingrediente_estandar', 'cantidad'], ascending=[True, True])

receta  \
14   Solomillo de vedella Strogonoff i arròs basmati   
27                                Cuinat de llenties   
74                                         Vinagreta   
272  Ous bullits amb salmó fumat, tàperes i maionesa   
608                                        Pa de lli   
..                                               ...   
603                                        Pa de lli   
672                                      Cetogranola   
3                            Iogurt grec amb fraules   
733                Iogurt grec amb ametlles i nabius   
898    Pudding de xía amb fruits vermells i ametlles   

            ingrediente_original  cantidad unidad ingrediente_estandar  
14                      Agua 10g       NaN    NaN                aigua  
27                 750ml d'aigua       NaN    NaN                aigua  
74                 10 ml d'aigua       NaN    NaN                aigua  
272                        Aigua       NaN    NaN                aigua  
608        120 g d’aigua calenta       NaN    NaN                aigua  
..                           ...       ...    ...                  ...  
603        xia (una culleradeta)      10.0      g                  xía  
672     5-10g de chía (opcional)      10.0      g                  xía  
3                     Xía 10-20g      20.0      g                  xía  
733               10-20g de chía      20.0      g                  xía  
898  3 cullerades soperes de xía      30.0      g                  xía  

[1063 rows x 5 columns]

In [154]:
df['ingrediente_estandar'] = df['ingrediente_estandar'].replace('salmo fresc', 'salmó fresc')
df['ingrediente_estandar'] = df['ingrediente_estandar'].replace('romani fresc', 'romaní fresc')


In [155]:
df.to_csv('ingredientes_estandarizados.csv', index=False)